src/api/shared-helpers/search-issues-cache.ts

```diff
@@ -0,0 +1,47 @@
+import { SearchIssueApiResponse, SearchIssueOptions } from '../types';
+
+interface SearchIssuesCache {
+	[ serializedParams: string ]: SearchIssueApiResponse;
+}
+
+interface SearchParams {
+	search: string;
+	options?: SearchIssueOptions;
+}
+
+const MAX_CACHE_SIZE = 25;
+
+let searchIssuesCache: SearchIssuesCache = {};
+let cacheKeyQueue: string[] = [];
+
+export function getSearchIssuesCache( {
+	search,
+	options,
+}: SearchParams ): SearchIssueApiResponse | undefined {
+	const serializedParams = serializeParams( { search, options } );
+	return searchIssuesCache[ serializedParams ];
+}
+
+export function setSearchIssuesCache(
+	{ search, options }: SearchParams,
+	response: SearchIssueApiResponse
+): void {
+	const serializedParams = serializeParams( { search, options } );
+	if ( cacheKeyQueue.length >= MAX_CACHE_SIZE ) {
+		const oldestCacheKey = cacheKeyQueue.shift();
```

Comment:
Hmmm, I'm pretty sure because of the `cacheKeyQueue.length >= MAX_CACHE_SIZE`, `oldestCacheKey` can't really ever be undefined! `shift()` only returns undefined if the array is empty.

The `if (oldestCacheKey)` check is mostly to keep the TypeScript transpiler happy, because it doesn't like the `string | undefined` type as a key indexer!

So I'm not entirely sure what extra error handling we should add here? Did you have something specific in mind?